gradle.properties

```diff
@@ -2,7 +2,7 @@
 org.gradle.jvmargs=-Duser.country=US -Dkotlin.daemon.jvm.options=-Xmx4096m -Dfile.encoding=UTF-8
 
 kotlinBaseVersion=2.1.0-dev-5441
-agpBaseVersion=7.3.1
+agpBaseVersion=8.0.2
```

Comment:
@ting-yuan To use Gradle 8, AGP 8 needs to be used. If KSP still needs to support AGP 7 there is a way, let me know.

Reply:
I'm a bit conservative on dropping support of older AGP versions. Can we stay with AGP 7 a bit longer?

Reply:
AGP 7 only supports Gradle 7 while AGP 8 only supports AGP 8.

If we want to we can try and change the Gradle runner version for the tests that use Android. But supporting multiple major versions might be tricky.

Reply:
So actually this isn't dropping support for older Android since no new APIs are being used. There is a test case for supporting AGP 7.4.1, I will add an additional one for 7.3.1

Reply:
+1, users can still use older AGP versions even if this gets upgraded.